packages/flutter/lib/src/material/text_field.dart

```diff
@@ -188,6 +188,54 @@ class _TextFieldSelectionGestureDetectorBuilder extends TextSelectionGestureDete
 /// ```
 /// {@end-tool}
 ///
+/// ## Reading values
+///
+/// A common way to read a value from a TextField is to use the [onSubmitted]
+/// method, which is called when the user finishes editing and receives the
+/// current value in its callback.
+///
+/// {@tool dartpad --template=stateless_widget_material}
```

Comment:
This should be a stateful widget and it should create the controller in initState() and dispose it in dispose().